sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -618,6 +622,11 @@ export class MessageSession extends LinkEntity<Receiver> {
    */
   subscribe(onMessage: OnMessage, onError: OnError, options?: MessageHandlerOptionsBase): void {
     if (!options) options = {};
+
+    if (options.abortSignal?.aborted) {
+      throw new AbortError(StandardAbortMessage);
+    }
```

Comment:
This seems to be the only net change... and is very deceptively simple...

I see how the abort signal is passed to the link creation time and the one with the `receiveMessages()`.
This change ensures that an already aborted signal will result in cancelling the subscribe() method.

How are we ensuring that once `subscribe()` is called and then abort signal is fired that we cancel the operation?

Reply:
Just to check - is this the scenario you're talking about with regards to cancellation?

https://github.com/Azure/azure-sdk-for-js/pull/11281/files#diff-220199656e7c07d2ec0c2ab9378ccb28R321

Reply:
Discussed offline, @richardpark-msft will be logging an issue for the concern I raised above

Reply:
(for summary)

The abortSignal passed into subscribe, currently, only applies logically to the initialization of the subscription itself. If the subscription starts (ie, handlers are registered for receiving messages, credits on the line) the abortSignal no longer applies.

The way we have it, as is, has tripped up a few people in discussions so we want to revisit this: https://github.com/Azure/azure-sdk-for-js/issues/11296